src/networking/spotify.api.ts

```diff
@@ -52,19 +52,19 @@ export class SpotifyApi {
             });
     }
 
-    getRecommendations = (genres?: string[], artists?: string[], retry = false): Promise<Track[]> => {
+    getRecommendations = (limit: number, genres?: string[], artists?: string[], retry = false): Promise<Track[]> => {
         const baseUrl = "https://api.spotify.com/v1/recommendations";
         const spotifyGenres = genres ? this.getSpotifyGenres(genres, true) : undefined;
         const queryParams = this.getQueryString({"seed_genres": spotifyGenres, "seed_artists": artists});
-        return fetch(`${baseUrl}?${queryParams}`, this.getHeaders())
+        return fetch(`${baseUrl}?${queryParams}&limit=${limit}`, this.getHeaders())
             .then(this.readResponse)
             .then(response => response.tracks)
             .then((tracks) => convertTracksToAppObject(tracks))
             .catch(() => {
                 if (retry) {
                     return Promise.reject("Error getting recommendations");
                 }
-                return this.onTokenExpired().then(() => this.getRecommendations(genres, artists, true));
+                return this.onTokenExpired().then(() => this.getRecommendations(limit, genres, artists, true));
```

Comment:
The limit param should be the third / last param of the recommendations function IMO